src/pgwire/src/protocol.rs

```diff
@@ -309,14 +309,35 @@ where
                     State::Ready => self.advance_ready().await?,
                     State::Drain => self.advance_drain().await?,
                     State::Done => return Ok(()),
-                }
+                };
+                self.adapter_client
```

Comment:
Oh this is interesting. The implementation is in SessionClient, but the caller has to opt in. Ok I like that.